src/App.tsx

```diff
@@ -15,36 +15,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SOTR_BY_ALPHABET: string = 'alphabet';
+const SOTR_BY_LENGTH: string = 'length';
+
+function SortGoodsFrom(
+  originArray: string[],
+  reverse: boolean,
+  sortBy: string,
+) {
+  let mutateArray: string[] = [...originArray];
+
+  switch (sortBy) {
+    case SOTR_BY_ALPHABET:
+      mutateArray = mutateArray.sort((a, b) => a.localeCompare(b));
```

Comment:
```suggestion
  const preparedGoods = [...originArray];

  switch (sortBy) {
    case SOTR_BY_ALPHABET:
      preparedGoods.sort((a, b) => a.localeCompare(b));
```
- Prefer const over let
- `arr/array/el/elem/obj/object/item/res/result` are bad names, name should describe a variable
- You don't need to assign a new value, `sort` mutates an array